learn/cookbooks/laravel-multitenancy.mdx

```diff
@@ -0,0 +1,365 @@
+
+First, let's install the dependencies:
+
+```bash
+npm install vue-instantsearch @meilisearch/instant-meilisearch
+```
+
+Now, let's create a Vue app that uses Vue InstantSearch. We will create a new `resources/js/vue-app.js` file:
+
+```js
+import { createApp } from 'vue'
+import InstantSearch from 'vue-instantsearch/vue3/es'
+import Meilisearch from './components/Meilisearch.vue'
+
+const app = createApp({
+  components: {
+    Meilisearch
+  }
+})
+
+app.use(InstantSearch)
+app.mount('#vue-app')
+```
+
+This file initializes our Vue app and configures it to use Vue InstantSearch. It also registers the `Meilisearch` component that we will create next.
+
+The `Meilisearch` component is responsible for initializing a Vue Instantsearch client. It uses the `@meilisearch/instant-meilisearch` package to create a search client compatible with Instantsearch.
+
+Let's create it in `resources/js/components/Meilisearch.vue`:
+
+```vue
+<script setup lang="ts">
+import { instantMeiliSearch } from "@meilisearch/instant-meilisearch"
+
+const props = defineProps<{
+  host: string,
+  apiKey: string,
+  indexName: string,
+}>()
+
+const { searchClient } = instantMeiliSearch(props.host, props.apiKey)
+</script>
+
+<template>
+  <ais-instant-search :search-client="searchClient" :index-name="props.indexName">
+    <!-- Slots allow you to render content inside this component, e.g. search results -->
+    <slot name="default"></slot>
+  </ais-instant-search>
+</template>
+```
+
+Now that we have our `Meilisearch` component, we can use it in any of our Blade views and provide it with the tenant token. Don't forget to add the `@vite` directive to include the Vue app in your view.
+
+```blade
+<!-- resources/views/contacts/index.blade.php -->
+
+<div id="vue-app">
+    <meilisearch index-name="contacts" api-key="{{ $meilisearchToken }}" host="https://edge.meilisearch.com">
+    </meilisearch>
+</div>
+
+@push('scripts')
+    @vite('resources/js/vue-app.js')
+@endpush
+```
+
+Et voilà! You now have a search interface that is secure and multitenant. Users can only access data from their organization, and you can rest assured that data from other tenants is safe.
+
+## Conclusion
+
+In this guide, we learned how to implement secure, multitenant search in a Laravel application. We generated tenant tokens for each organization and used them to secure access to Meilisearch. We also built a search interface using Vue InstantSearch and provided it with the tenant token.
+
+All the code in this guide is a simplified example of what we implemented in our [Laravel CRM](https://saas.meilisearch.com/?utm_campaign=oss&utm_source=docs&utm_medium=laravel-multitenancy) example application. You can find the full code in our [GitHub repository](https://github.com/meilisearch/saas-demo).
```

Comment:
Daniele Procida says it's very arrogant to assume people have learnt things after reading your documentation 😡

```suggestion
In this guide, you saw how to implement secure, multitenant search in a Laravel application. You then generated tenant tokens for each organization and used them to secure access to Meilisearch. You also built a search interface using Vue InstantSearch and provided it with the tenant token.

All the code in this guide is a simplified example of what we implemented in the [Laravel CRM](https://saas.meilisearch.com/?utm_campaign=oss&utm_source=docs&utm_medium=laravel-multitenancy) example application. Find the full code on [GitHub](https://github.com/meilisearch/saas-demo).
```